content/tutorials/numpy_integration/grass_numpy_integration.qmd

```diff
@@ -0,0 +1,253 @@
+  ipynb: default
+  html:
+    toc: true
+    code-tools: true
+    code-copy: true
+    code-fold: false
+engine: jupyter
+execute:
+  eval: false
+jupyter: python3
+---
+
+# Introduction
+
+This short tutorial shows how to integrate NumPy arrays with GRASS tools to create a smooth workflow for scientific modeling and analysis in Python.
+
+Specifically, it demonstrates a complete workflow: generating a terrain surface in GRASS, importing it into [Landlab](https://landlab.csdms.io/) (an open-source Python library for running numerical models of Earth-surface dynamics), running an erosion model, and then returning the results to GRASS for further hydrologic analysis.
+
+With the [grass.tools API](https://grass.osgeo.org/grass-devel/manuals/python_intro.html) (introduced in GRASS v8.5), tools can now accept and return NumPy arrays directly, rather than working only with GRASS rasters. While native rasters remain the most efficient option for large-scale processing, NumPy arrays make it easy to connect GRASS with the broader Python scientific ecosystem, enabling advanced analysis and seamless use of familiar libraries.
+
+::: {.callout-note title="How to run this tutorial?"}
+
+This tutorial is prepared to be run in a Jupyter notebook locally
+or using services such as Google Colab. You can [download the notebook here](grass_numpy_integration.ipynb).
+
+If you are not sure how to get started with GRASS, checkout the tutorials [Get started with GRASS & Python in Jupyter Notebooks](../get_started/fast_track_grass_and_python.qmd)
+and [Get started with GRASS in Google Colab](../get_started/grass_gis_in_google_colab.qmd).
+
+:::
+
+![](logos.webp)
+
+# Generating a fractal surface in GRASS
+
+First we will import GRASS packages:
+
+```{python}
+# import standard Python packages
+import os
+import sys
+import subprocess
+
+sys.path.append(
+    subprocess.check_output(["grass", "--config", "python_path"], text=True).strip()
+)
+
+# import GRASS python packages
+import grass.script as gs
+import grass.jupyter as gj
+from grass.tools import Tools
+```
+
+Create a new GRASS project called "landlab". Since we will be working with artifical data, we don't need to
+provide any coordinate reference system, resulting in a generic cartesian system.
+
+```{python}
+gs.create_project("landlab")
+```
+
+Initialize a session in this project and create a `Tools` object we will use for calling GRASS tools:
+
+```{python}
+session = gj.init("landlab")
+tools = Tools()
+```
+
+Since we are generating artificial data, we need to specify the dimensions (number of rows and columns).
+We also need to let GRASS know the actual coordinates; we will do that by setting
+the [computational region](https://grass.osgeo.org/grass-stable/manuals/g.region.html). Lower-left (south-west) corner of the data will be at the coordinates (0, 0),
+the coordinates of the upper-right (nort-east) corner are number of rows times cell resolution and number of columns times cell resolution.
```

Comment:
The resolution choice is not clear here. What about reiterating what happened after the call and mentioning there that "we picked the resolution..." Or just cut that out or make it some extra point in the text.

Reply:
It may be worth mentioning that while some processes don't care, more advanced uses require resolution, like xy_spacing=resolution below. I guess a relevant GRASS example would be the hydrology.